src/wallet/rpcwallet.cpp

```diff
@@ -2749,16 +2749,17 @@ UniValue listwallets(const JSONRPCRequest& request)
             "Returns a list of currently loaded wallets.\n"
             "For full information on the wallet, use \"getwalletinfo\"\n"
             "\nResult:\n"
-            "[                         (json array of strings)\n"
-            "  \"walletname\"            (string) the wallet name\n"
-            "   ...\n"
-            "]\n"
+            "{\n"
+            "  \"loaded\":     [ \"walletname\" ... ] (json array of strings) Names of loaded wallets\n"
+            "  \"available\":  [ \"walletname\" ... ] (json array of strings) Names of BDB files in wallet directory (may not always actually be wallets)\n"
```

Comment:
Would suggest returning an array of objects rather than an array of strings to future proof this. This would allow including other information like sizes, last modification times, whether databases are flushed, or locked by another process. (Latter two would be meaningful after #11687 when wallet files can be opened in their own berkelydb environments instead of all using the same shared environment).

Reply:
Partially agree with @ryanofsk. Still think one array is enough, see https://github.com/bitcoin/bitcoin/pull/11485#issuecomment-335813034. And `listwallets` could have an optional parameter to filter available, loaded or both.